src/main/java/io/jenkins/plugins/credentials/secretsmanager/AwsCredentialsProvider.java

```diff
@@ -28,17 +49,16 @@
     private final AwsCredentialsStore store = new AwsCredentialsStore(this);
 
     private final Supplier<Collection<IdCredentials>> credentialsSupplier =
-            CredentialsSupplierFactory.create();
+            memoizeWithExpiration(AwsCredentialsProvider::fetchCredentials, Duration.ofMinutes(5));
```

Comment:
Curious where the 5 minutes value comes from. Would it make sense to make it configurable at some point?

Reply:
5 minutes is IIRC the recommended default in the Credentials Provider implementor’s guide. Making it user-configurable is indeed on the to do list: [JENKINS-57577](https://issues.jenkins-ci.org/browse/JENKINS-57577)